libraries/chain/db_maint.cpp

```diff
@@ -925,7 +925,22 @@ void database::perform_chain_maintenance(const signed_block& next_block, const g
    // Reset all BitAsset force settlement volumes to zero
    for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
    {
-      modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      if(head_block_time() < HARDFORK_CORE_518_TIME)
+         modify( d, [](asset_bitasset_data_object& o) { o.force_settled_volume = 0; });
+      else { // clean expired feeds
+         modify( d, [this](asset_bitasset_data_object& o) {
+            o.force_settled_volume = 0;
+            // Remove all expired feeds
+            for( auto itr = o.feeds.begin(); itr != o.feeds.end();)
```

Comment:
Please use a reverse iterator for better performance while erasing multiple entries.